tests/integration/events.test.js

```diff
@@ -1014,6 +1019,112 @@ describe('EVENTS ROUTES', () => {
+
+        const response = await app.inject({
+          method: 'POST',
+          url: '/events',
+          payload,
+          headers: { Cookie: `alenvi_token=${authToken}` },
+        });
+
+        expect(response.statusCode).toEqual(200);
+        expect(response.result.data.event).toBeDefined();
+      });
+
+      it('should return 403 if absence extension and extended absence are not for the same valid reason', async () => {
+        const payload = {
+          type: ABSENCE,
+          startDate: '2019-01-23T10:00:00',
+          endDate: '2019-01-23T12:30:00',
+          auxiliary: eventsList[20].auxiliary,
+          absenceNature: DAILY,
+          absence: MATERNITY_LEAVE,
+          extension: eventsList[20]._id,
+        };
+
+        const response = await app.inject({
+          method: 'POST',
+          url: '/events',
+          payload,
+          headers: { Cookie: `alenvi_token=${authToken}` },
+        });
+
+        expect(response.statusCode).toEqual(403);
+      });
+
+      it('should return 403 if extended absence nature is HOURLY', async () => {
+        const payload = {
+          type: ABSENCE,
+          startDate: '2019-01-23T10:00:00',
+          endDate: '2019-01-23T12:30:00',
+          auxiliary: eventsList[20].auxiliary,
+          absenceNature: HOURLY,
+          absence: PARENTAL_LEAVE,
+          extension: eventsList[20]._id,
+        };
+
+        const response = await app.inject({
+          method: 'POST',
+          url: '/events',
+          payload,
+          headers: { Cookie: `alenvi_token=${authToken}` },
+        });
+
+        expect(response.statusCode).toEqual(403);
+      });
+
+      it('should return 403 if extended absence reason is invalid', async () => {
+        const payload = {
+          type: ABSENCE,
+          startDate: '2019-01-23T10:00:00',
+          endDate: '2019-01-23T12:30:00',
+          auxiliary: eventsList[1].auxiliary,
+          absenceNature: DAILY,
+          absence: PAID_LEAVE,
+          extension: eventsList[1]._id,
+        };
+
+        const response = await app.inject({
+          method: 'POST',
+          url: '/events',
+          payload,
+          headers: { Cookie: `alenvi_token=${authToken}` },
+        });
+
+        expect(response.statusCode).toEqual(403);
+      });
+
+      it('should return 403 if event startDate is before extended absence startDate', async () => {
+        const payload = {
+          type: ABSENCE,
+          startDate: '2019-01-17T10:00:00',
+          endDate: '2019-01-17T12:30:00',
+          auxiliary: eventsList[20].auxiliary,
+          absenceNature: DAILY,
+          absence: PARENTAL_LEAVE,
+          extension: eventsList[20]._id,
+        };
+
+        const response = await app.inject({
+          method: 'POST',
+          url: '/events',
+          payload,
+          headers: { Cookie: `alenvi_token=${authToken}` },
+        });
+
+        expect(response.statusCode).toEqual(403);
+      });
```

Comment:
should return a 403 if trying to extend an absence already extended once

Reply:
je ne comprends pass ton commentaire Felix